pylinalg/func/vector.py

```diff
@@ -116,3 +116,114 @@ def vector_apply_matrix(vectors, matrix, /, *, w=1, out=None, dtype=None):
     if dtype is not None:
         out = out.astype(dtype, copy=False)
     return out
+
+
+def project_camera(vector, projection_matrix) -> np.ndarray:
+    """Project a 3D vector into 2D space.
+
+    Parameters
+    ----------
+    vector : ArrayLike
+        The vector to be projected.
+    projection_matrix: ArrayLike
+        The camera's intrinsic matrix.
+
+    Returns
+    -------
+    projected_vector : ndarray
+        The projection of the vector.
+
+    """
+
+    raise NotImplementedError()
+
+
+def unproject_camera(vector, projection_matrix, /, *, depth=0) -> np.ndarray:
+    """Un-project a vector from 2D space to 3D space.
```

Comment:
unlike `project_camera` this function is more interesting than a simple matrix transformation, since there is the `depth` parameter, so this seems useful enough to keep around.